src/SimulatedEffect/Http.elm

```diff
@@ -0,0 +1,322 @@
+        }
+        |> SimulatedEffect.Task
+
+
+{-| An HTTP header for configuring requests.
+-}
+type alias Header =
+    ( String, String )
+
+
+{-| Create a `Header`.
+-}
+header : String -> String -> Header
+header =
+    Tuple.pair
+
+
+{-| Represents the body of a `Request`.
+-}
+type Body
+    = EmptyBody
+    | StringBody
+        { contentType : String
+        , content : String
+        }
+
+
+{-| Create an empty body for your `Request`.
+-}
+emptyBody : Body
+emptyBody =
+    EmptyBody
+
+
+{-| Put some JSON value in the body of your `Request`. This will automatically
+add the `Content-Type: application/json` header.
+-}
+jsonBody : Json.Encode.Value -> Body
+jsonBody value =
+    StringBody
+        { contentType = "application/json"
+        , content = Json.Encode.encode 0 value
+        }
+
+
+{-| Put some string in the body of your `Request`.
+-}
+stringBody : String -> String -> Body
+stringBody contentType content =
+    StringBody
+        { contentType = contentType
+        , content = content
+        }
+
+
+{-| Logic for interpreting a response body.
+-}
+type Expect msg
+    = Expect (Http.Response String -> msg)
+
+
+{-| Expect the response body to be a `String`.
+-}
+expectString : (Result Http.Error String -> msg) -> Expect msg
+expectString onResult =
+    Expect <|
+        \response ->
+            case response of
+                Http.BadUrl_ s ->
+                    onResult (Err <| Http.BadUrl s)
+
+                Http.Timeout_ ->
+                    onResult (Err Http.Timeout)
+
+                Http.NetworkError_ ->
+                    onResult (Err Http.NetworkError)
+
+                Http.BadStatus_ metadata body ->
+                    onResult (Err <| Http.BadStatus metadata.statusCode)
+
+                Http.GoodStatus_ _ body ->
+                    onResult (Ok body)
+
+
+{-| Expect the response body to be JSON.
+-}
+expectJson : (Result Http.Error a -> msg) -> Decoder a -> Expect msg
+expectJson onResult decoder =
+    Expect <|
+        \response ->
+            case response of
+                Http.BadUrl_ s ->
+                    onResult (Err <| Http.BadUrl s)
+
+                Http.Timeout_ ->
+                    onResult (Err Http.Timeout)
+
+                Http.NetworkError_ ->
+                    onResult (Err Http.NetworkError)
+
+                Http.BadStatus_ metadata _ ->
+                    onResult (Err <| Http.BadStatus metadata.statusCode)
+
+                Http.GoodStatus_ _ body ->
+                    case Json.Decode.decodeString decoder body of
+                        Err jsonError ->
+                            onResult (Err <| Http.BadBody <| Json.Decode.errorToString jsonError)
+
+                        Ok value ->
+                            onResult (Ok value)
+
+
+{-| Expect the response body to be whatever.
+-}
+expectWhatever : (Result Error () -> msg) -> Expect msg
+expectWhatever onResult =
```

Comment:
When do we want whatever?

Reply:
This is just paralleling the function in elm/http: https://package.elm-lang.org/packages/elm/http/latest/Http#expectWhatever  (but produces a SimulatedEffect.Http.Expect instead of an Http.Expect) -- the full docs are in elm/http